File: PRID5853.G06.Entities/Models/Comment.cs
using PRID5853.G06.Entities.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace PRID5853.G06.Entities.Models
{
    public class Comment : EntityBase
    {
        [Required]
        public string Body { get; set; }
        public DateTime Timestamp { get; set; }
        public DateTime? EditTimeStamp { get; set; }
        public virtual User Author { get; set; }
        public int AuthorId { get; set; }
        public virtual Post Post { get; set; }
        public int PostId { get; set; }
    }
}

File: PRID5853-G06/AngularClient/ClientApp/src/app/components/models/comment.ts
import { Author } from "./Author";

export class Comment {
    id: number;
    body: string;
    timestamp: Date;
    editTimeStamp : Date;
    postId: number;
    authorId: number;
    author : Author;


    constructor(id = 0, body = "", timestamp = new Date(), editTimeStamp = null, postId = 0, authorId = 0, author = null){
        this.id = id;
        this.body = body;
        this.timestamp = timestamp;
        this.editTimeStamp = editTimeStamp;
        this.postId = postId;
        this.authorId = authorId;
        this.author = author;
    }
}